[BUG] MockLogAppender appended to after stopping

See test failure here: https://build.ci.opensearch.org/job/gradle-check/28539/consoleText

```
org.opensearch.transport.netty4.SimpleNetty4TransportTests > testTracerLog FAILED
    java.lang.AssertionError: 
    Expected: an empty collection
         but: <[Attempted to append to non-started appender mock]>
        at __randomizedtesting.SeedInfo.seed([C98A4BF58C821570:CA09B64B8D8ABBA5]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:964)
        at org.junit.Assert.assertThat(Assert.java:930)
        at org.opensearch.test.OpenSearchTestCase.checkStaticState(OpenSearchTestCase.java:638)
        at org.opensearch.test.OpenSearchTestCase.after(OpenSearchTestCase.java:428)
        at java.****/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.****/java.lang.reflect.Method.invoke(Method.java:578)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at java.****/java.lang.Thread.run(Thread.java:1623)

```

I thought I fixed this problem in #1595. After glancing at the code I don't see where the problem is so this remains a mystery.

--- a/server/src/test/java/org/opensearch/common/logging/LoggersTests.java
+++ b/server/src/test/java/org/opensearch/common/logging/LoggersTests.java
@@ -83,11 +83,11 @@ public class LoggersTests extends OpenSearchTestCase {
             assertThat(appender.lastParameterizedMessage().getParameters(), arrayContaining(Arrays.asList("yes", "no")));
 
             ex = randomException();
-            testLogger.trace(() -> new ParameterizedMessage("a trace message; element = [{}]", new Object[]{null}), ex);
+            testLogger.trace(() -> new ParameterizedMessage("a trace message; element = [{}]", new Object[] { null }), ex);
             assertThat(appender.lastEvent.getLevel(), equalTo(Level.TRACE));
             assertThat(appender.lastEvent.getThrown(), equalTo(ex));
             assertThat(appender.lastParameterizedMessage().getFormattedMessage(), equalTo("a trace message; element = [null]"));
-            assertThat(appender.lastParameterizedMessage().getParameters(), arrayContaining(new Object[]{null}));
+            assertThat(appender.lastParameterizedMessage().getParameters(), arrayContaining(new Object[] { null }));
         } finally {
             Loggers.removeAppender(testLogger, appender);
             appender.stop();

--- a/server/src/test/java/org/opensearch/common/logging/LoggersTests.java
+++ b/server/src/test/java/org/opensearch/common/logging/LoggersTests.java
@@ -53,40 +53,45 @@ public class LoggersTests extends OpenSearchTestCase {
         appender.start();
         final Logger testLogger = LogManager.getLogger(LoggersTests.class);
         Loggers.addAppender(testLogger, appender);
-        Loggers.setLevel(testLogger, Level.TRACE);
+        try {
+            Loggers.setLevel(testLogger, Level.TRACE);
 
-        Throwable ex = randomException();
-        testLogger.error(() -> new ParameterizedMessage("an error message"), ex);
-        assertThat(appender.lastEvent.getLevel(), equalTo(Level.ERROR));
-        assertThat(appender.lastEvent.getThrown(), equalTo(ex));
-        assertThat(appender.lastParameterizedMessage().getFormattedMessage(), equalTo("an error message"));
+            Throwable ex = randomException();
+            testLogger.error(() -> new ParameterizedMessage("an error message"), ex);
+            assertThat(appender.lastEvent.getLevel(), equalTo(Level.ERROR));
+            assertThat(appender.lastEvent.getThrown(), equalTo(ex));
+            assertThat(appender.lastParameterizedMessage().getFormattedMessage(), equalTo("an error message"));
 
-        ex = randomException();
-        testLogger.warn(() -> new ParameterizedMessage("a warn message: [{}]", "long gc"), ex);
-        assertThat(appender.lastEvent.getLevel(), equalTo(Level.WARN));
-        assertThat(appender.lastEvent.getThrown(), equalTo(ex));
-        assertThat(appender.lastParameterizedMessage().getFormattedMessage(), equalTo("a warn message: [long gc]"));
-        assertThat(appender.lastParameterizedMessage().getParameters(), arrayContaining("long gc"));
+            ex = randomException();
+            testLogger.warn(() -> new ParameterizedMessage("a warn message: [{}]", "long gc"), ex);
+            assertThat(appender.lastEvent.getLevel(), equalTo(Level.WARN));
+            assertThat(appender.lastEvent.getThrown(), equalTo(ex));
+            assertThat(appender.lastParameterizedMessage().getFormattedMessage(), equalTo("a warn message: [long gc]"));
+            assertThat(appender.lastParameterizedMessage().getParameters(), arrayContaining("long gc"));
 
-        testLogger.info(() -> new ParameterizedMessage("an info message a=[{}], b=[{}], c=[{}]", 1, 2, 3));
-        assertThat(appender.lastEvent.getLevel(), equalTo(Level.INFO));
-        assertThat(appender.lastEvent.getThrown(), nullValue());
-        assertThat(appender.lastParameterizedMessage().getFormattedMessage(), equalTo("an info message a=[1], b=[2], c=[3]"));
-        assertThat(appender.lastParameterizedMessage().getParameters(), arrayContaining(1, 2, 3));
+            testLogger.info(() -> new ParameterizedMessage("an info message a=[{}], b=[{}], c=[{}]", 1, 2, 3));
+            assertThat(appender.lastEvent.getLevel(), equalTo(Level.INFO));
+            assertThat(appender.lastEvent.getThrown(), nullValue());
+            assertThat(appender.lastParameterizedMessage().getFormattedMessage(), equalTo("an info message a=[1], b=[2], c=[3]"));
+            assertThat(appender.lastParameterizedMessage().getParameters(), arrayContaining(1, 2, 3));
 
-        ex = randomException();
-        testLogger.debug(() -> new ParameterizedMessage("a debug message options = {}", Arrays.asList("yes", "no")), ex);
-        assertThat(appender.lastEvent.getLevel(), equalTo(Level.DEBUG));
-        assertThat(appender.lastEvent.getThrown(), equalTo(ex));
-        assertThat(appender.lastParameterizedMessage().getFormattedMessage(), equalTo("a debug message options = [yes, no]"));
-        assertThat(appender.lastParameterizedMessage().getParameters(), arrayContaining(Arrays.asList("yes", "no")));
+            ex = randomException();
+            testLogger.debug(() -> new ParameterizedMessage("a debug message options = {}", Arrays.asList("yes", "no")), ex);
+            assertThat(appender.lastEvent.getLevel(), equalTo(Level.DEBUG));
+            assertThat(appender.lastEvent.getThrown(), equalTo(ex));
+            assertThat(appender.lastParameterizedMessage().getFormattedMessage(), equalTo("a debug message options = [yes, no]"));
+            assertThat(appender.lastParameterizedMessage().getParameters(), arrayContaining(Arrays.asList("yes", "no")));
 
-        ex = randomException();
-        testLogger.trace(() -> new ParameterizedMessage("a trace message; element = [{}]", new Object[] { null }), ex);
-        assertThat(appender.lastEvent.getLevel(), equalTo(Level.TRACE));
-        assertThat(appender.lastEvent.getThrown(), equalTo(ex));
-        assertThat(appender.lastParameterizedMessage().getFormattedMessage(), equalTo("a trace message; element = [null]"));
-        assertThat(appender.lastParameterizedMessage().getParameters(), arrayContaining(new Object[] { null }));
+            ex = randomException();
+            testLogger.trace(() -> new ParameterizedMessage("a trace message; element = [{}]", new Object[]{null}), ex);
+            assertThat(appender.lastEvent.getLevel(), equalTo(Level.TRACE));
+            assertThat(appender.lastEvent.getThrown(), equalTo(ex));
+            assertThat(appender.lastParameterizedMessage().getFormattedMessage(), equalTo("a trace message; element = [null]"));
+            assertThat(appender.lastParameterizedMessage().getParameters(), arrayContaining(new Object[]{null}));
+        } finally {
+            Loggers.removeAppender(testLogger, appender);
+            appender.stop();
+        }
     }
 
     private Throwable randomException() {

--- a/server/src/test/java/org/opensearch/common/settings/InsecureSettingTests.java
+++ b/server/src/test/java/org/opensearch/common/settings/InsecureSettingTests.java
@@ -51,12 +51,12 @@ public class InsecureSettingTests extends OpenSearchTestCase {
             }
         };
         rootAppender.start();
-        Loggers.addAppender(LogManager.getRootLogger(), rootAppender);
+        Loggers.addAppender(LogManager.getLogger(SecureSetting.class), rootAppender);
     }
 
     @After
     public void removeInsecureSettingsAppender() {
-        Loggers.removeAppender(LogManager.getRootLogger(), rootAppender);
+        Loggers.removeAppender(LogManager.getLogger(SecureSetting.class), rootAppender);
         rootAppender.stop();
     }
 

--- a/server/src/test/java/org/opensearch/common/settings/InsecureSettingTests.java
+++ b/server/src/test/java/org/opensearch/common/settings/InsecureSettingTests.java
@@ -25,7 +25,7 @@ public class InsecureSettingTests extends OpenSearchTestCase {
     private List<String> rootLogMsgs = new ArrayList<>();
     private AbstractAppender rootAppender;
 
-    protected void assertSettingWarning() {
+    private void assertSettingWarning() {
         assertWarnings(
             "[setting.name] setting was deprecated in OpenSearch and will be removed in a future release! See the breaking changes documentation for the next major version."
         );
@@ -50,13 +50,14 @@ public class InsecureSettingTests extends OpenSearchTestCase {
                 InsecureSettingTests.this.rootLogMsgs.add(message);
             }
         };
-        Loggers.addAppender(LogManager.getRootLogger(), rootAppender);
         rootAppender.start();
+        Loggers.addAppender(LogManager.getRootLogger(), rootAppender);
     }
 
     @After
     public void removeInsecureSettingsAppender() {
         Loggers.removeAppender(LogManager.getRootLogger(), rootAppender);
+        rootAppender.stop();
     }
 
     public void testShouldRaiseExceptionByDefault() {

